src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/DataController.cs

```diff
@@ -97,15 +96,26 @@ public async Task<ActionResult<DataElement>> Delete(int instanceOwnerPartyId, Gu
                 return dataElementError;
             }
 
-            string storageFileName = DataElementHelper.DataFileName(instance.AppId, instanceGuid.ToString(), dataGuid.ToString());
+            if (delay)
+            {
+                // consideration: what happens if you try to delete an already harddeleted dataElement? 404? 
+                (Application application, ActionResult applicationError) = await GetApplicationAsync(instance.AppId, instance.Org);
```

Comment:
consideration: what happens if you try to delete an already harddeleted dataElement? 404?

Reply:
Yeah, 404 is the logical choice. The resource doesn't exist.

Reply:
Another consideration.. 404 makes sense for the end user. What do we do if the appOwner is attempting to delete the dataElement. Cause in their case the element is still exposed?